### DISCUSSION TITLE
[init_implicit] private or protected copy construtor.

### DISCUSSION THREAD

**Discussion by huangweiwu** [wjakob/nanobind#270] [Q&A] [answered]
When a subclass has a private or protected copy construtor, it can't use init_implicit to convert to it's baseclass.
The compile error info:'B::B': cannot access private member declared in class 'B'".
Is there any way to avoid this without modifying class B.
```
class A {};

class B : public A
{
private:
    int a;
    B(const B&);
};


NB_MODULE(my_ext, m) {
    nb::class_<A>(m, "A")
        .def(nb::init_implicit<B>())
        ;
    nb::class_<B>(m, "B")
        ;
}

**Comment by wjakob** [comment, marked as answer]
You are missing an ampersand (`&`) in the `init_implicit` declaration. Otherwise you are binding a call-by-value constructor, and your type is not copyable.

**Reply by huangweiwu** [reply]
Get it!
